content/docs/serverless-workflows/troubleshooting.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Troubleshooting
+date: "2024-07-25"
+weight: 101
+---
+
+
+# Troubleshooting Guide
+
+This document provides solutions to common problems encountered with serverless workflows.
+
+## Table of Contents
+
+1. [Workflow Errors](#workflow-errors)
+2. [Configuration Problems](#configuration-problems)
+3. [Performance Issues](#performance-issues)
+4. [Error Messages](#error-messages)
+5. [Network Problems](#network-problems)
+6. [Common Scenarios](#common-scenarios)
+7. [Contact Support](#contact-support)
+
+---
+
+## Workflow Errors
+
+### Problem: Workflow execution fails
+
+**Solution:**
+
+1. Examine the container log of the workflow
+    ```console
+        kubectl logs my-workflow-xy73lj
+    ```
+
+### Problem: Workflow is not listed by the orchestrator plugin
+
+**Solution:**
+
+1. Examine the container log of the workflow
+    ```console
+        kubectl logs my-workflow-xy73lj
+    ```
+
+2. Most probably the Data index service was unready when the workflow started.
+   Typically this is what the log shows:
+    ```console
+        2024-07-24 21:10:20,837 ERROR [org.kie.kog.eve.pro.ReactiveMessagingEventPublisher] (main) Error while creating event to topic kogito-processdefinitions-events for event ProcessDefinitionDataEvent {specVersion=1.0, id='586e5273-33b9-4e90-8df6-76b972575b57', source=http://mtaanalysis.default/MTAAnalysis, type='ProcessDefinitionEvent', time=2024-07-24T21:10:20.658694165Z, subject='null', dataContentType='application/json', dataSchema=null, data=org.kie.kogito.event.process.ProcessDefinitionEventBody@7de147e9, kogitoProcessInstanceId='null', kogitoRootProcessInstanceId='null', kogitoProcessId='MTAAnalysis', kogitoRootProcessId='null', kogitoAddons='null', kogitoIdentity='null', extensionAttributes={kogitoprocid=MTAAnalysis}}: java.util.concurrent.CompletionException: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: sonataflow-platform-data-index-service.default/10.96.15.153:80 
+    ```
+
+3. Make sure the Data Index is ready, and restart the workflow
+    ```console
+        $ oc get pods sonataflow-platform-data-index-service-546f59f89f-b7548 
```

Comment:
let's maintain consistency by the cli used in this guide (either `oc` or `kubectl`, but only one of them)

we should also specify the namespace of the platform. If the orchestrator is installed by the OCP chart or the operator, then the destination should be the `sonataflow-infra` namespace.

I'm not concerned about advanced users that creates their own data-index by adding platform per-namespace explicitly.

Reply:
good catch, in orchestrator-k8s it is per ns